substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/phases/IntrinsifyMethodHandlesInvocationPlugin.java

```diff
@@ -487,10 +490,15 @@ private void processInvokeWithMethodHandle(GraphBuilderContext b, Replacements r
              */
             Node singleFunctionality = null;
             ReturnNode singleReturn = null;
+            ValueNode singleNewInstance = null;
             for (Node node : graph.getNodes()) {
                 if (node == graph.start() || node instanceof ParameterNode || node instanceof ConstantNode || node instanceof FrameState) {
                     /* Ignore the allowed framework around the nodes we care about. */
-                } else if (node instanceof MethodCallTargetNode) {
+                } else if (node instanceof DynamicNewInstanceNode && singleNewInstance == null && ((DynamicNewInstanceNode) node).getInstanceType().isConstant()) {
```

Comment:
Why is such a  `DynamicNewInstanceNode` not canonicalized to a `NewInstanceNode`? Because the type is not initialized yet?

Reply:
I don't know for sure.  This was the initial cause of the failure requiring the fixes.

Reply:
I only see the DynamicNewInstanceNodes during the clinit phase of native-image.